### PR TITLE
Test wrapper component

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

File: src/modules/wrapper/wrapper.component.ts [hunk 1]
```diff
@@ -43,14 +44,15 @@ export class StacheWrapperComponent implements OnInit, AfterContentInit {
...
     this.titleService.setTitle(this.windowTitle || this.pageTitle);
 
     this.route.fragment.subscribe(fragment => {
-      setImmediate(() => {
-        const element = document.getElementById(fragment);
+      this.windowRef.nativeWindow.setImmediate(() => {
+        const element = this.windowRef.nativeWindow.document.getElementById(fragment);
         if (element) {
           element.scrollIntoView();
         }
```
Comment (Blackbaud-SteveBrush) [inline src/modules/wrapper/wrapper.component.ts]: We need a newline at the end of the file.

File: src/modules/wrapper/wrapper.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+import { expect } from '@blackbaud/skyux-builder/runtime/testing/browser';
+import { SkyAppWindowRef } from '@blackbaud/skyux-builder/runtime';
+
+import { StacheWrapperComponent } from './wrapper.component';
+import { StacheLayoutComponent } from '../layout/layout.component';
+import { StacheWrapperTestComponent } from './fixtures/wrapper.component.fixture';
+import { StacheTitleService } from './title.service';
+
+describe('Stache Wrapper Component', () => {
```
Comment (Blackbaud-SteveBrush) [inline src/modules/wrapper/wrapper.component.spec.ts]: Our other specs don't include spaces between the words `StacheWrapperComponent`.

File: src/modules/wrapper/wrapper.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    TestBed.configureTestingModule({
+      declarations: [
+        StacheLayoutComponent,
+        StacheWrapperComponent,
+        StacheWrapperTestComponent
+      ],
+      providers: [
+        { provide: ActivatedRoute, useValue: mockActivatedRoute },
+        { provide: Router, useValue: mockRouter },
```
Comment (Blackbaud-SteveBrush) [inline src/modules/wrapper/wrapper.component.spec.ts]: Both `mockActivatedRoute` and `mockRouter` are not being used in the tests, so you can instead inject the class itself into the providers:

```
{ provide: ActivatedRoute, useClass: MockActivatedRoute }
```

File: karma.config.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      noInfo: true,
+      stats: 'minimal'
+    },
+    reporters: ['mocha', 'coverage'],
+    port: 9876,
+    colors: true,
+    logLevel: config.LOG_INFO,
+    autoWatch: false,
+    singleRun: true,
+    browserNoActivityTimeout: 100000
```
Comment (Blackbaud-SteveBrush) [inline karma.config.js]: @blackbaud-brandonhare This line is what made it work. Apparently, the wrapper tests are taking quite a while and Chrome timed out. IMO we have two options: 1) Double-check the tests to make sure the code isn't caught in a time warp, or 2) submit a PR to Builder to add the `browserNoActivityTimeout` to the test config.

Comment (Blackbaud-SteveBrush) [inline karma.config.js]: FYI, I added the karma.config.js file purely for debugging purposes.

**Comment by Blackbaud-SteveBrush** [comment]
Also, code coverage is at 100%, but the branches are near 87.5%.

File: karma.config.js (hunk 4 above)
Comment (blackbaud-brandonhare) [inline karma.config.js]: If it's caught in a loop, why wouldn't it time out on our machines? That's strange. I believe it was working previously with `setImmediate`, but I don't want to go back to using a non standard window method if `setInterval` achieves the same result.

Comment (blackbaud-brandonhare) [inline karma.config.js]: The other strange thing is that we are mocking `setTimeout`. It's not actually calling the native method. So I'm not sure why that would affect it.

File: package.json [hunk 5]
```diff
@@ -10,7 +10,8 @@
     "node": ">=6.9.1"
   },
   "scripts": {
-    "skyux": "skyux"
+    "skyux": "skyux",
+    "test": "node --max-old-space-size=4096 ./node_modules/karma/bin/karma start ./karma.config.js --command test"
```
Comment (Blackbaud-SteveBrush) [inline package.json]: Sorry man, I didn't remove this from yesterday's "experiment". Mind?

File: src/modules/wrapper/wrapper.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    component.pageTitle = 'Page Title';
+    fixture.detectChanges();
+    expect(mockTitleService.setTitle).toHaveBeenCalledWith('Page Title');
+  });
+
+  it('should grab the element from the fragment', async(() => {
+    component.ngOnInit();
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+    expect(mockWindowService.nativeWindow.document.getElementById)
```
Comment (Blackbaud-SteveBrush) [inline src/modules/wrapper/wrapper.component.spec.ts]: Needs a tab

**Pull request merged**